### ISSUE TITLE
new check failing -> scanner/registration.go:21 +0x311

### ISSUE THREAD

**Issue by louwersj** [tfsec/tfsec#731]
Technically not a bug.

Trying to extend tfsec by creating a new set of checks for another cloud provider.. I have cloned aws001.go and crafted a new check as a first attempt. After some initial issues coded a valid check and the build is working as expected. However, when executing I run into the below error. 

`goroutine 1 [running]:
github.com/tfsec/tfsec/internal/app/tfsec/scanner.RegisterCheck(0x96b123, 0x6, 0x98d5a8, 0x3e, 0x99e570, 0xb7, 0x98db40, 0x3f, 0x97eb1e, 0x22, ...)
	/vagrant/src/github.com/tfsec/tfsec/internal/app/tfsec/scanner/registration.go:21 +0x311
github.com/tfsec/tfsec/internal/app/tfsec/checks.init.119()
	/vagrant/src/github.com/tfsec/tfsec/internal/app/tfsec/checks/oci001.go:33 +0x178`

This makes me believe I have to register a new check (not being AWS, Azure or GCP) in a somewhat more extensive way. Any guidance on where (and how) this should be done so I can start creating additional checks for Oracle Cloud and try to extend the functionality of tfsec?

**Comment by owenrumney** [comment]
Hey @louwersj, really appreciate the new provider checks.

You should just be able to do a RegisterChecks - could create a PR or point me to a branch which your check so I can see what might be the issue?

**Comment by louwersj** [comment]
@owenrumney Looks like the changes you did solved the problem.... I crafted a very very (very very) simple example case to show the first check for OCI to scan for the use of an IP from a public IP pool instead of using one from your private IP pool. 

This was just an example to show the working and I guess we have passed this one. Which in effect would open the way for us to start building a more extensive set of checks for Oracle Cloud :) 

`Check 4

  [OCI001][WARNING] Resource 'opc_compute_ip_address_reservation.my-ip-address' is using an IP from a public IP pool
  /vagrant/src/github.com/tfsec/tfsec/bin/linux/test2.tf:16


      13 | 
      14 | resource "opc_compute_ip_address_reservation" "my-ip-address" {
      15 |   name            = "my-ip-address"
      16 |   ip_address_pool = "public-ippool"
      17 | }
      18 | 
      19 | resource "opc_compute_ip_network" "my-ip-network" {

  Impact:     The compute instance has the ability to be reached from outside
  Resolution: Reconsider the use of an public IP

  See https://tfsec.dev/docs/oracle/OCI001/ for more information. 
`
